packages/sui-domain/src/EntryPointFactory.js

```diff
@@ -42,13 +43,34 @@ export default ({useCases, config}) =>
 
       // if loader is undefined then is not implemented, otherwhise load async the useCase
       return loader === undefined
-        ? new NotImplementedUseCase(key)
+        ? createNotImplementedUseCase(key)
         : {
             execute: params => {
+              const subscriptionsForUseCase = this.subscribers[key] || []
               // load async the factory, execute use case and return the promise
-              return loader().then(factory =>
-                getMethod(factory)({config: this._config}).execute(params)
-              )
+              return loader()
+                .then(factory =>
+                  getMethod(factory)({config: this._config}).execute(params)
+                )
+                .then(result => {
+                  subscriptionsForUseCase.forEach(fn =>
+                    fn({error: null, params, result})
+                  )
+                  return result
+                })
+                .catch(e => {
+                  subscriptionsForUseCase.forEach(fn =>
+                    fn({error: e, params, result: null})
+                  )
+                  return Promise.reject(e)
+                })
+            },
+            subscribe: callback => {
+              this.subscribers[key] = this.subscribers[key] || []
+              this.subscribers[key].push(callback)
+
+              // return a way to remove the listener
+              return () => this.subscribers[key].filter(l => l !== callback)
```

Comment:
As a subjective opinion, I like more an `rxjs` API where you can get a subscription with a `unsubscribe` method.
Getting directly the `unsubscribe` function when `subscribe` has been executed can cause confusion to me.
Some `rxjs` docs: https://rxjs-dev.firebaseapp.com/guide/subscription

Reply:
Then we could do something like this. The drawback I see is that we will lose the possibility of cleaning the effect without storing a variable.

```
useEffect(() => {
    const {unsubscribe} = domain.get('generate_search_url_search_use_case').subscribe()
// ...
    return unsubscribe
  }, [])
```